packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -0,0 +1,186 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/widgets.dart';
+
+import 'flat_button.dart';
+import 'icon_button.dart';
+import 'icons.dart';
+import 'material.dart';
+import 'theme.dart';
+
+const double _kHandleSize = 22.0; // pixels
+const double _kToolbarScreenPadding = 8.0; // pixels
+
+/// Manages a copy/paste text selection toolbar.
+class _TextSelectionToolbar extends StatelessWidget {
+  _TextSelectionToolbar(this.delegate, {Key key}) : super(key: key);
+
+  final TextSelectionDelegate delegate;
+  InputValue get value => delegate.inputValue;
+
+  @override
+  Widget build(BuildContext context) {
+    List<Widget> items = <Widget>[];
+
+    if (!value.selection.isCollapsed) {
+      items.add(new FlatButton(child: new Text('CUT'), onPressed: _handleCut));
+      items.add(new FlatButton(child: new Text('COPY'), onPressed: _handleCopy));
+    }
+    items.add(new FlatButton(
+      child: new Text('PASTE'),
+      onPressed: delegate.pasteBuffer != null ? _handlePaste : null)
+    );
+    if (value.selection.isCollapsed) {
+      items.add(new FlatButton(child: new Text('SELECT ALL'), onPressed: _handleSelectAll));
+    }
+    // TODO(mpcomplete): implement `more` menu.
+    items.add(new IconButton(icon: Icons.more_vert));
+
+    return new Material(
+      elevation: 1,
+      child: new Container(
```

Comment:
Hmm.. when I try to use a ButtonBar, there is too much padding, and the `more` menu is placed too far down vertically. With Row, there is less padding than there used to be, but it is still too much.